### PR TITLE
AlmaLinux/build-system#24: Add mechanism of generating community PGP keys

### PR DESCRIPTION
**Pull request by soksanichenko**
- The subject is implemented
- Some parameters are migrated from str to pathlib.Path
- Refactored

### CODE REVIEW

**Review by Korulag — APPROVED** [review]

File: sign_node/signer.py [hunk 1]
```diff
@@ -29,15 +34,27 @@
...
+
+gpg_scenario_template = (
+    '%no-protection\n'
+    'Key-Type: RSA\n'
+    'Key-Length: 2048\n'
+    'Subkey-Type: default\n'
+    'Subkey-Length: 2048\n'
+    'Name-Real: {sign_key_uid}\n'
+    'Expire-Date: 0\n'
+)
```
Comment (m10k) [inline sign_node/signer.py:57]: 2048 bits might be a bit short, especially if the keys never expire

**Review by m10k — APPROVED** [review]

**Review by zklevsha — APPROVED** [review]

File: sign_node/signer.py [hunk 2]
```diff
@@ -114,61 +132,131 @@ def on_sync_request(self, queue, message):
...
-                **{'clear-sign': '--clear-sign',
-                   'detach-sign': '--detach-sign'})
+                **{
+                    'clear-sign': '--clear-sign',
+                    'detach-sign': '--detach-sign',
+                }
+            )
             with tempfile.NamedTemporaryFile(mode='w') as fd:
-                asc_file_name = f'{fd.name}.asc'
+                asc_file_name = Path(fd.name).with_suffix('.asc')
```
Comment (anfimovdm) [inline sign_node/signer.py:141]: you can declare suffix `.asc` in `tempfile.NamedTemporaryFile`
```python3
with tempfile.NamedTemporaryFile(mode='w', suffix='.asc') as fd:
    ...
```

File: sign_node/signer.py [hunk 3]
```diff
@@ -16,8 +16,13 @@
 import typing
 import urllib.parse
 from collections import defaultdict
-from concurrent.futures import ThreadPoolExecutor, as_completed
+from concurrent.futures import (
+    ThreadPoolExecutor,
+    as_completed,
+    )
```
Comment (anfimovdm) [inline sign_node/signer.py:22]: indent

File: sign_node/signer.py (hunk 1 above)
Comment (anfimovdm) [reply sign_node/signer.py:57]: can we move it to the constants?

File: sign_node/signer.py [hunk 4]
```diff
@@ -114,61 +132,131 @@ def on_sync_request(self, queue, message):
...
             with tempfile.NamedTemporaryFile(mode='w') as fd:
-                asc_file_name = f'{fd.name}.asc'
+                asc_file_name = Path(fd.name).with_suffix('.asc')
                 fd.write(payload['content'])
                 fd.flush()
                 sign_cmd = plumbum.local['gpg'][
-                    '--yes', sig_type[payload.get('sig_type', 'detach-sign')],
-                    '--armor', '--default-key', payload['key_id'], fd.name
+                    '--yes',
+                    sig_type[payload.get('sig_type')],
```
Comment (anfimovdm) [inline sign_node/signer.py:146]: why we don't use `detach-sign` as default value in `get`?

File: sign_node/signer.py [hunk 5]
```diff
@@ -114,61 +132,131 @@ def on_sync_request(self, queue, message):
...
+        self._report_signed_build(task['id'], response_payload)
+
+    def report_generate_sign_key_error(self, task: typing.Dict, msg: str):
+        sign_key_name = self._generate_key_uid(task)
+        response_payload = {
+            'key_name': sign_key_name,
+            'success': False,
+            'error_message': msg,
+        }
+        logging.info('Test debug message 2')
```
Comment (anfimovdm) [inline sign_node/signer.py:196]: I think we don't need that

File: sign_node/signer.py [hunk 6]
```diff
@@ -114,61 +132,131 @@ def on_sync_request(self, queue, message):
...
+                task_id = task['id']
+                try:
+                    processing_method(task)
+                    logging.info('The task "%s" is processed', task_id)
+                except Exception as err:
+                    logging.exception(
+                        'Can\'t process task from web server because "%s"',
+                        err,
+                    )
+                    logging.info('Test debug message')
```
Comment (anfimovdm) [inline sign_node/signer.py:245]: the same as above

File: sign_node/signer.py [hunk 7]
```diff
@@ -222,6 +310,122 @@ def check(pkg_path: str) -> typing.Tuple[SignStatusEnum, str]:
...
+
+    @staticmethod
+    def _extract_key_fingerprint(keyid: str) -> str:
+        fingerprint_cmd = plumbum.local['gpg'][
+            '-k',
+            keyid,
+        ]
+        _, stdout, _ = fingerprint_cmd.run()
+
+        key_fingerprint = stdout.split('\n')[1].strip()
```
Comment (anfimovdm) [inline sign_node/signer.py:326]: please left a comment why we take second element from array

**Review by anfimovdm — CHANGES_REQUESTED** [review]

File: sign_node/signer.py (hunk 1 above)
Comment (soksanichenko) [reply sign_node/signer.py]: Resolved

File: sign_node/signer.py (hunk 2 above)
Comment (soksanichenko) [reply sign_node/signer.py]: Thanks!

File: sign_node/signer.py (hunk 3 above)
Comment (soksanichenko) [reply sign_node/signer.py]: Done

File: sign_node/signer.py (hunk 4 above)
Comment (soksanichenko) [reply sign_node/signer.py]: Done

File: sign_node/signer.py (hunk 6 above)
Comment (soksanichenko) [reply sign_node/signer.py]: Removed

File: sign_node/signer.py (hunk 7 above)
Comment (soksanichenko) [reply sign_node/signer.py:323]: Done, thank you

**Pull request merged**